test/signatures.test.ts

```diff
@@ -0,0 +1,134 @@
+import {ethers} from 'ethers';
+import {hashState, State} from '../src/contract/state';
+import {AddressZero} from 'ethers/constants';
+import {arrayify, splitSignature} from 'ethers/utils';
+import {getStateSignerAddress, signChallengeMessage, signState} from '../src/signatures';
+
+describe('signatures', () => {
+  describe('signState', () => {
+    it('signs a state', async () => {
+      const wallet = ethers.Wallet.createRandom();
+      const state: State = {
+        channel: {chainId: '0x1', channelNonce: '0x1', participants: [wallet.address]},
+        outcome: [],
+        turnNum: 1,
+        isFinal: false,
+        appData: '0x0',
+        appDefinition: AddressZero,
+        challengeDuration: '0x5',
+      };
+
+      const signedState = await signState(state, wallet.privateKey);
+
+      expect(signedState).toMatchObject({
+        state,
+        signature: {
+          r: expect.any(String),
+          s: expect.any(String),
+          v: expect.any(Number),
+        },
+      });
+
+      expect(getStateSignerAddress(signedState)).toEqual(wallet.address);
+    });
+    it('throws an exception if signing with non-participant private key', async () => {
+      const wallet = ethers.Wallet.createRandom();
+      const state: State = {
```

Comment:
These test cases can likely be shortened if we add a default test state constructor that returns a state object with reasonable defaults.